matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/pushrules/ProcessEventForPushTask.kt

```diff
@@ -73,6 +75,7 @@ internal class DefaultProcessEventForPushTask @Inject constructor(
                         " to check for push rules with ${params.rules.size} rules"
         )
         val matchedEvents = allEvents.mapNotNull { event ->
+            eventDecryptor.decryptEventAndSaveResult(event, "")
```

Comment:
This might do unnecessary redecryption. Should be a decrypt if needed (test if amready decrypted before retrying).
Also at this point if it's not decrypted, I don't think that the second attempt will work. We don't have the key.

Reply:
`decryptEventAndSaveResult` just returns if the event is already decrypted.
I just tested again with some logs and I confirm that the event is successfully decrypted at this moment:
```
13:03:51.024  V  [PushRules] --> checkPushRules
13:03:51.043  V  Execute transaction in 20 millis
13:03:51.063  V  [PushRules] Found 1 out of 1 to check for push rules with 27 rules
13:03:51.064  D  decryptEventAndSaveResult | event.getClearType()=m.room.encrypted
13:03:51.065  V  decryptEvent $i8Dx2Ri7FIr-meu********, requestKeysOnFail:true
13:03:51.065  V  ## Inbound: getInboundGroupSession  2Ge4kbC0P******** in cache true
13:03:51.069  V  ##########################################################
13:03:51.070  V  ## decryptGroupMessage() timeline: 
13:03:51.070  V  ## decryptGroupMessage() senderKey: AvjFBpOE************
13:03:51.071  V  ## decryptGroupMessage() sessionId: 2Ge4k***************
13:03:51.071  V  ## decryptGroupMessage() roomId: !fngDRO********
13:03:51.072  V  ## decryptGroupMessage() eventId: $i8Dx2Ri7FIr-meu********
13:03:51.072  V  ## decryptGroupMessage() mIndex: 17
13:03:51.073  V  ## dispatchLiveEventDecrypted $i8Dx2Ri7FIr-meu******
13:03:51.075  D  decryptEventAndSaveResult | update event with decryption result: MXEventDecryptionResult(clearEvent={room_id=!fngDRO******, type=m.room.message, content={msgtype=m.text, body=test event in encrypted room}}, senderCurve25519Key=AvjFBpOEsqVexQ9F********, claimedEd25519Key=R5fd2pJs9XH******, forwardingCurve25519KeyChain=[], isSafe=true)
13:03:51.105  D  Skipping notification update due to event list not changing
13:03:51.139  V  [PushRules] Rule PushRule(actions=[notify, {set_tweak=sound, value=default}], default=false, enabled=true, ruleId=.m.rule.message, conditions=[PushCondition(kind=event_match, key=type, pattern=m.room.message, iz=null)], pattern=null) match for event $i8Dx2Ri7FIr-meu*******
13:03:51.140  D  [PushRules] matched 1 out of 1
13:03:51.141  V  [PushRules] Found 0 redacted events
13:03:51.143  D  Push rule match for event $i8Dx2Ri7FIr-meu******
13:03:51.148  V  [PushRules] <-- Push task scheduled
```

I agree that this is probably not the best place to try to decrypt but it seems relevant to do the decryption. But I don't really know where it'd be better to do the decryption tbh

Reply:
I think I see the problem.
In DefaultProcessEventForPushTask there is that:
> it.takeIf { ... }?.copy(roomId = key)

This is doing a copy to add the roomId to the event, but the normal copy is not copying @Transiant fields. So it removed the decryption result.

So I think it should use `copyAll()`instead.